src/formatting/cargo_toml.rs

```diff
@@ -0,0 +1,412 @@
+use itertools::Itertools;
+use std::cmp::Ordering;
+use toml_edit::{
+    visit_mut::*, Decor, Document, Formatted, Item, KeyMut, Table, TableLike, TomlError, Value,
+};
+
+use crate::{Config, ErrorKind};
+
+pub(crate) fn format_cargo_toml_inner(content: &str, config: &Config) -> Result<String, ErrorKind> {
+    let mut doc = content.parse::<toml_edit::Document>()?;
+    let rules: Vec<Box<dyn VisitMut>> = vec![
+        Box::new(SortSection {
+            current_position: 0,
+        }),
+        Box::new(SortKey),
+        Box::new(BlankLine { trimming: true }),
+        Box::new(KeyValue),
+        Box::new(MultiLine),
+        Box::new(WrapArray {
+            max_width: config.max_width(),
+        }),
+        Box::new(FormatInlineTable {
+            max_width: config.max_width(),
+            long_tables: vec![],
+            current_section: String::new(),
+        }),
+    ];
```

Comment:
I don't think any of the boxes are necessary here; shouldn't just adding `&mut` to the beginning of the values and replacing `Box<dyn VisitMut>` with `&mut dyn VisitMut` suffice?